microprofile/tests/testng/src/main/java/io/helidon/microprofile/tests/testng/HelidonTestNGListener.java

```diff
@@ -0,0 +1,509 @@
+            resetPerTest = testAnnot.resetPerTest();
+        }
+
+        DisableDiscovery discovery = testClass.getAnnotation(DisableDiscovery.class);
+        if (discovery != null) {
+            classLevelDisableDiscovery = discovery.value();
+        }
+
+        if (resetPerTest) {
+            validatePerTest();
+            return;
+        }
+        validatePerClass();
+
+        configure(classLevelConfigMeta);
+
+        if (!classLevelConfigMeta.useExisting) {
+            startContainer(classLevelBeans, classLevelExtensions, classLevelDisableDiscovery);
+        }
+
+        //Injecting Instances to CDI
+        Object[] instances = iTestClass.getInstances(false);
+
+        for (Object instance : instances) {
+            injectTestToCDI(instance, iTestClass.getRealClass());
+        }
+    }
+
+
+    @Override
+    public void onAfterClass(ITestClass testClass) {
+        if (!resetPerTest) {
+            releaseConfig();
+            stopContainer();
+        }
+    }
+
+    @Override
+    public void onTestStart(ITestResult result) {
+
+        if (resetPerTest) {
+            Method method = result.getMethod().getConstructorOrMethod().getMethod();
+            AddConfig[] configs = method.getAnnotationsByType(AddConfig.class);
+            ConfigMeta methodLevelConfigMeta = classLevelConfigMeta.nextMethod();
+            methodLevelConfigMeta.addConfig(configs);
+            methodLevelConfigMeta.configuration(method.getAnnotation(Configuration.class));
+
+            configure(methodLevelConfigMeta);
+
+            List<AddExtension> methodLevelExtensions = new ArrayList<>(classLevelExtensions);
+            List<AddBean> methodLevelBeans = new ArrayList<>(classLevelBeans);
+            boolean methodLevelDisableDiscovery = classLevelDisableDiscovery;
+
+            AddExtension[] extensions = method.getAnnotationsByType(AddExtension.class);
+            methodLevelExtensions.addAll(Arrays.asList(extensions));
+
+            AddBean[] beans = method.getAnnotationsByType(AddBean.class);
+            methodLevelBeans.addAll(Arrays.asList(beans));
+
+            DisableDiscovery discovery = method.getAnnotation(DisableDiscovery.class);
+            if (discovery != null) {
+                methodLevelDisableDiscovery = discovery.value();
+            }
+
+            startContainer(methodLevelBeans, methodLevelExtensions, methodLevelDisableDiscovery);
+        }
+    }
+
+    @Override
+    public void onTestFailure(ITestResult iTestResult) {
+        if (resetPerTest) {
+            releaseConfig();
+            stopContainer();
+        }
+    }
+
+    @Override
+    public void onTestSuccess(ITestResult iTestResult) {
+        if (resetPerTest) {
+            releaseConfig();
+            stopContainer();
+        }
+    }
+
+    /*
+     * Helper method to inject TestNG initialized classes into CDI Container.
+     */
+    private <T> void injectTestToCDI(Object bean, final Class<T> clazz) {
```

Comment:
Use camel case even for abbreviations like CDI (e.g. the method name should be `injectTestToCdi`) - this does not require a change, but it must be followed for public API

Reply:
renamed